deployments/Dockerfiles/tendermint/tendermint.py

```diff
@@ -194,22 +197,25 @@ def _start_monitoring_thread(self) -> None:
 
     def _stop_tm_process(self) -> None:
         """Stop a Tendermint node process."""
-        if self._process is None:
+
+        process = self._process
+        if process is None:
             return
 
-        if platform.system() == "Windows":
-            os.kill(self._process.pid, signal.CTRL_C_EVENT)  # type: ignore  # pylint: disable=no-member
-            try:
-                self._process.wait(timeout=5)
-            except subprocess.TimeoutExpired:  # nosec
-                os.kill(self._process.pid, signal.CTRL_BREAK_EVENT)  # type: ignore  # pylint: disable=no-member
-        else:
-            self._process.send_signal(signal.SIGTERM)
-            self._process.wait(timeout=5)
-            poll = self._process.poll()
-            if poll is None:  # pragma: nocover
-                self._process.terminate()
-                self._process.wait(3)
+        process.poll()
+        if process.returncode is not None:
+            return
+
+        send_control_c(process)
+        try:
+            process.wait(timeout=TERMINATION_TIMEOUT)
+        except subprocess.TimeoutExpired:
+            if platform.system() == "Windows":
+                process.send_signal(signal.CTRL_BREAK_EVENT)  # type: ignore  # pylint: disable=no-member
+                process.wait(timeout=TERMINATION_TIMEOUT)
+            else:
+                process.terminate()
+                process.wait(timeout=TERMINATION_TIMEOUT)
```

Comment:
specifically the wait after sending the `signal.CTRL_BREAK_EVENT` should address the issue with the hard reset on Windows